subsys/bluetooth/mesh/net.c

```diff
@@ -285,37 +287,37 @@ bool bt_mesh_net_iv_update(uint32_t iv_index, bool iv_update)
 		return false;
 	}
 
-	if ((iv_index > bt_mesh.iv_index + 1) ||
-	    (iv_index == bt_mesh.iv_index + 1 &&
-	     (atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS) || !iv_update))) {
+	/* Also treats the IV Flag as an diffed value as well as IV Index */
+	diffs = (iv_index - bt_mesh.iv_index) +
+		(iv_update == atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS));
```

Comment:
Honestly struggling to see this expression any better than the previous one. The current one easy to read and map to the specification.

Reply:
The current approach also treats the IV Flag as an diffed value as well as IV Index, and if it is greater than 1,
then the Recovery occurs.

To me, this logic is simpler